pkgs/development/libraries/pipewire/media-session.nix

```diff
@@ -48,13 +48,32 @@ let
 
     mesonFlags = [
       "-Ddocs=enabled"
+      "-Dsystemd-system-service=enabled"
       # We generate these empty files from the nixos module, don't bother installing them
       "-Dwith-module-sets=[]"
     ];
 
     postUnpack = ''
       patchShebangs source/doc/input-filter-h.sh
       patchShebangs source/doc/input-filter.sh
+
+      # Don't try to put system service units into pkgs.systemd.
+      # The definition streches across two lines, that's why we delete the
+      # second one.
+      sed -Ei "
+        s@^(systemd_system_services_dir).*@\\1 = '$out/lib/systemd/system'@
+        /^\s+pkgconfig_define:/d
+      " source/systemd/system/meson.build
+
+      # Build pipewire-media-session.service unconditionally.
+      # Without this, the build will fail with:
+      #   ERROR: Tried to access unknown option "session-managers".
+      # pipewire-media-session doesn't have a "session-managers" option so
+      # it seems as if it wans to access pkgs.pipewire, which does have it
+      # and sets it to the empty string.
+      sed -Ei "
+        s@if get_option.*@if true@
+      " source/systemd/system/meson.build
```

Comment:
Sounds like a omission when it was split out. Could you please submit a MR upstream?

Reply:
This has already been fixed upstream in [dfa7401](https://gitlab.freedesktop.org/pipewire/media-session/-/commit/dfa740175c83e1cd0d815ad423f90872de566437). How to proceed here? I'd like to keep that substitution for now, and add a comment to the fixing commit and a notice that this can be removed probably once pipewire-media-session releases something >0.4.1.

Reply:
maybe we reference it in patches with a note to drop it on >0.4.1 ? it will probably not apply anymore and cause a build failure anyway

Reply:
I like the idea of directly referencing the patch and letting builds fail as soon as they are not applicable anymore. I've done the same for [fixing `rootprefix`](https://gitlab.freedesktop.org/pipewire/media-session/-/commit/2ff6b0baec7325dde229013b9d37c93f8bc7edee.diff) after creating [an upstream PR]( https://gitlab.freedesktop.org/pipewire/media-session/-/merge_requests/28).